File: typescript/behavioral/template_method/guru_example/concreteClass2.ts
import {AbstractClass} from "./abstractClass";

export class ConcreteClass2 extends AbstractClass {
  protected requiredOperation1() {
    console.log('ConcreteClass2: requiredOperation1.')
  }

  protected requiredOperation2() {
    console.log('ConcreteClass2: requiredOperation2.')
  }

  protected hook1() {
    console.log('ConcreteClass2: hook1.')
  }
}

File: dart/behavioral/template_method/guru_example/bin/concrete_class2.dart
import 'abstract_class.dart';

class ConcreteClass2 extends AbstractClass {
  @override
  void requiredOperation1() {
    print('ConcreteClass2: requiredOperation1.');
  }

  @override
  void requiredOperation2() {
    print('ConcreteClass2: requiredOperation2.');
  }

  @override
  void hook1() {
    print('ConcreteClass2: hook1.');
  }
}
